### PR TITLE
test(AWSDataStoreCategoryPluginFlutterIntegrationTests): FlutterSerializedModel tests

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
Adds suite of integration tests consuming the FlutterSerializedModel APIs.

This PR replaces this [older version](https://github.com/aws-amplify/amplify-ios/pull/1440), which was closed due to a bad rebase. 

*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginFlutterIntegrationTests/TestSupport/AmplifyFlutterHelpers/FlutterSerializedModel.swift [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+import Foundation
+import Amplify
+
+/// `FlutterSerializedModel` assists in serializing and deserializing JSON when interacting with the amplify-ios DataStore API.
+/// Taken from [amplify-flutter DataStore category](https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_datastore/ios/Classes/types/model/FlutterSerializedModel.swift)
+struct FlutterSerializedModel: Model, JSONValueHolder {
```
Comment (lawmicha) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginFlutterIntegrationTests/TestSupport/AmplifyFlutterHelpers/FlutterSerializedModel.swift:12]: Will this ever change in Flutter? If so, have you thought about how we should maintain this class in both places? ie. Do we need a comment in the Flutter codebase to make sure if modified, we should change and run integration test in iOS?

Comment (haverchuck) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginFlutterIntegrationTests/TestSupport/AmplifyFlutterHelpers/FlutterSerializedModel.swift:12]: @lawmicha I don't think the validity of these tests would be impacted by a change to FlutterSerializedModel, but I still think your suggestion for [a comment in the flutter codebase](https://github.com/aws-amplify/amplify-flutter/pull/1192) is a good one.

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginFlutterIntegrationTests/Connection/DataStoreConnectionScenario7FlutterTests.swift [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+ type Comment6 @model @key(name: "byPost", fields: ["postID", "content"]) {
+   id: ID!
+   postID: ID!
+   post: Post6 @connection(fields: ["postID"])
+   content: String!
+ }
+ ```
+ */
+
+class DataStoreConnectionScenario6FlutterTests: SyncEngineFlutterIntegrationTestBase {
```
Comment (thisisabhash) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginFlutterIntegrationTests/Connection/DataStoreConnectionScenario7FlutterTests.swift:39]: please rename the file to `DataStoreConnectionScenario6FlutterTests.swift`

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginFlutterIntegrationTests/Connection/DataStoreConnectionScenario2FlutterTests.swift [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+            case .failure(let error):
+                XCTFail("failed \(error)")
+            }
+        }
+        wait(for: [completeInvoked], timeout: TestCommonConstants.networkTimeout)
+        return result
+    }
+}
+
+extension Team2: Equatable {
```
Comment (thisisabhash) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginFlutterIntegrationTests/Connection/DataStoreConnectionScenario2FlutterTests.swift:385]: line 385 to 397 can be removed

**Comment by thisisabhash** [comment]
I ran the integration tests in my local machine and they were all passing.
The build also succeeds using SPM.

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginFlutterIntegrationTests/TestSupport/AmplifyFlutterHelpers/FlutterDataStoreRequestUtils.swift [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        guard let modelSchema = modelSchemas[modelName] else {
+            throw DataStoreError.decodingError(
+                "ModelSchema for Model \(modelName) is not registered",
+                "Ensure all the ModelSchemas are correctly registered")
+        }
+        return modelSchema
+    }
+    
+    static func getSerializedModelData(methodChannelArguments: [String: Any]) throws -> [String: Any] {
+        guard let serializedModelData = methodChannelArguments["serializedModel"] as? [String: Any] else {
```
[the remark this answers is not in the document]
Comment (thisisabhash) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginFlutterIntegrationTests/TestSupport/AmplifyFlutterHelpers/FlutterDataStoreRequestUtils.swift]: `guard let serializedModelData = methodChannelArguments["serializedData"] as? [String: Any] else {`

something like this? 

i see the key being "serializedData" at  line 218 in FlutterSerializedModel.swift
```
public func toMap(modelSchema: ModelSchema) -> [String: Any] {
        return [
            "id": id,
            "modelName": modelSchema.name,
            "serializedData": generateSerializedData(modelSchema: modelSchema)
        ]
    }
```

**Review by thisisabhash — APPROVED** [review]
LGTM

**Pull request merged**